zebra-state/src/service/memory_state/non_finalized_state.rs

```diff
@@ -182,3 +185,94 @@ impl NonFinalizedState {
             .map(|box_chain| box_chain.deref())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use zebra_chain::serialization::ZcashDeserializeInto;
+    use zebra_test::prelude::*;
+
+    use crate::tests::FakeChainHelper;
+
+    use self::assert_eq;
+    use super::*;
+
+    #[test]
+    fn best_chain_wins() -> Result<()> {
+        zebra_test::init();
+
+        let block2: Arc<Block> =
+            zebra_test::vectors::BLOCK_MAINNET_419201_BYTES.zcash_deserialize_into()?;
+        let block1: Arc<Block> =
+            zebra_test::vectors::BLOCK_MAINNET_419200_BYTES.zcash_deserialize_into()?;
+        // Create a random block which will have a much worse difficulty hash
```

Comment:
Work is calculated from the difficulty threshold field, not the hash.
```suggestion
        // Create a random block which will have a much worse difficulty threshold
```